### DISCUSSION TITLE
401 error accessing https://api.bentley.com/imodelhub/sv1.1/Repositories/iModel--<id>/iModelScope/Briefcase/

### DISCUSSION THREAD

**Discussion by PeterJBassett** [iTwin/itwinjs-core#2975] [Q&A] [answered]
Hello - I am running an integration test of my imodel-bridge application (FutureOn Connector), and am getting the following error:

POST 401 https://api.bentley.com/imodelhub/sv1.1/Repositories/iModel--ff1caedc-a986-4972-8a6e-b3f58707062a/iModelScope/Briefcase/ (658ms)

The integration test uses the @bentley/oidc-signin-tool and it looks like this is using the URL https://imsoidc.bentley.com/as/4rRu9/resume/as/authorization.ping (358)
Whereas I thought that this URL is no longer used and it should be https://ims.bentley.com?

Is this the reason for the 401 error? and if so, what is the solution?

**Comment by aruniverse** [comment]
Have you tried the new [@itwin/oidc-signin-tool](https://www.npmjs.com/package/@itwin/oidc-signin-tool)? It should point to ims.bentley.com.

**Comment by PeterJBassett** [comment]
Thank you for your reply.
Unfortunately @itwin/oidc-signin-tool has a quite different interface to the previous @bentley/oidc-signin-tool which makes it difficult to try without re-writing the integration test.
Are there any simpler solutions?

**Comment by PeterJBassett** [comment]
Hello - As requested, I have updated my test to use @itwin/oidc-signin-tool .
The test signs in ok, but when the test runs the synchronisation, it gives the error "[Error: download manifest failed (1) - SSL peer certificate or SSH remote key was not OK]"
The test is a node.js script, I have tried node v14 and node v16, but still get the same error.
I am running the test on a Macbook with MacOS Monterey
The test uses BridgeRunner from @bentley/imodel-bridge

Do you know the solution to the error?

**Reply by aruniverse** [reply]
@jchick-bentley ? Wonder if this is related, https://github.com/iTwin/itwinjs-core/discussions/2862, and you need to do something similar

**Reply by PeterJBassett** [reply]
Thank you for your reply. Yes I think it is related, it is the same problem but on a different platform.
The solution on https://github.com/iTwin/itwinjs-core/discussions/2862 was for a docker container. Now I have the same problem running on a Macbook, so it needs a different solution that will solve the problem on a Macbook.

**Reply by jchick-bentley** [reply]
Hi Peter,

Unfortunately, I am not familiar with this error and not a Macbook user.  Do you have any more details about the error?  i.e. Log messages/output surrounding the error message.  I'll pose the question to my colleagues who may have experience on MacOS.

Jason

**Comment by PeterJBassett** [comment, marked as answer]
Here are some more details:
The code where the error occurs is:

Silent sign-in:

      const { TestUsers, TestUtility } = require('@itwin/oidc-signin-tool')

      // Silent sign-in to Bentley and get access token to run a synchronisation
      const iModelAccessTokenString = await TestUtility.getAccessToken(TestUsers.regular)

The successfully returns a JWT access token.

Run the synchronisation:

      console.log("Running the Synchronisation")
      const fwk = new BridgeRunner(bridgeJobDef, serverArgs);
      const status = await fwk.synchronize();

This is using the following modules:

import { BentleyStatus, Config } from "@bentley/bentleyjs-core";
import { IModelHostConfiguration, IModelHost } from "@bentley/imodeljs-backend";
import { BridgeRunner, BridgeJobDefArgs} from "@bentley/imodel-bridge";
import { ServerArgs } from "@bentley/imodel-bridge/lib/IModelHubUtils";
import { AccessToken, AuthorizedClientRequestContext } from "@bentley/itwin-client";
import { IModelJsConfig } from "@bentley/config-loader/lib/IModelJsConfig";

The module versions are 2.9.18

The only message output is:

DevTools listening on ws://127.0.0.1:54798/devtools/browser/7d5df2ff-4ec7-42aa-8541-b712137d600e
[0106/103808.160769:INFO:CONSOLE(358)] "isWebAuthnPlatformAuthenticatorAvailable - false", source: https://ims.bentley.com/as/7Mz6E/resume/as/authorization.ping (358)

[0106/114000.792156:ERROR:command_buffer_proxy_impl.cc(122)] ContextResult::kTransientFailure: Failed to send GpuChannelMsg_CreateCommandBuffer.
[0106/114000.793104:ERROR:command_buffer_proxy_impl.cc(122)] ContextResult::kTransientFailure: Failed to send GpuChannelMsg_CreateCommandBuffer.

[Error: download manifest failed (1) - SSL peer certificate or SSH remote key was not OK]

That is all I have got to go on.

**Reply by jchick-bentley** [reply]
@PeterJBassett I created a new GitHub issue for this (ref. https://github.com/iTwin/itwinjs-core/issues/2997).  We intend to fix it in versions 2.19.x and 3.x and will update this discussion when the fix is available.  Would you be willing to test it for us?

**Reply by PeterJBassett** [reply]
Hello Jason Thank you for your reply. Yes of course I will be happy to try the new version.

**Reply by aruniverse** [reply]
Issue was resolved, selecting this thread as the answer to close it out
